subsys/bluetooth/services/ots/otc.c

```diff
@@ -169,15 +126,91 @@ static int read_attr(struct bt_conn *conn,
 		    struct bt_otc_internal_instance_t *inst,
 		    uint16_t handle, bt_gatt_read_func_t cb);
 
-static void print_oacp_response(enum bt_ots_oacp_proc_type req_opcode,
-				enum bt_ots_oacp_res_code result_code)
+/* L2CAP callbacks */
+static void tx_done (struct bt_gatt_ots_l2cap *l2cap_ctx,
+		     struct bt_conn *conn)
+{
+	/* Not doing any writes yet */
+	BT_ERR("Unexpected call, context: %p, conn: %p", l2cap_ctx, conn);
+}
+
+static ssize_t rx_done (struct bt_gatt_ots_l2cap *l2cap_ctx,
+			 struct bt_conn *conn, struct net_buf *buf)
+{
+	uint32_t offset = cur_inst->rcvd_size;
+	bool is_complete = false;
+	struct bt_otc_obj_metadata *cur_object =
+		&cur_inst->otc_inst->cur_object;
+	int cb_ret;
+
+	BT_DBG("Incoming L2CAP data, context: %p, conn: %p, len: %u, offset: %u",
+	       l2cap_ctx, conn, buf->len, offset);
+
+	cur_inst->rcvd_size += buf->len;
+
+	if (cur_inst->rcvd_size >= cur_object->current_size) {
+		is_complete = true;
+	}
+
+	if (cur_inst->rcvd_size > cur_object->current_size) {
+		BT_WARN("Received %u but expected maximum %u",
+			cur_inst->rcvd_size, cur_object->current_size);
+	}
```

Comment:
Alternatively leave the latter `if` as is, and change `is_complete` to 
```
const bool is_complete = cur_inst->rcvd_size >= cur_object->current_size;
```

Reply:
I can not see that this would do any functional change, so I am afraid I might be missing something.
Is the issue that the two if-s are partly "overlapping", and that you find it cleaner to have the second as a "specialization" of the first?